src/TicketBai/Mews.Fiscalizations.TicketBai/ModelToDtoConverter.cs

```diff
@@ -0,0 +1,395 @@
+            return new EncadenamientoFacturaAnteriorType
+            {
+                SerieFacturaAnterior = originalInvoiceInfo.Series.Map(s => s.Value).GetOrNull(),
+                NumFacturaAnterior = originalInvoiceInfo.Number.Value,
+                FechaExpedicionFacturaAnterior = Convert(originalInvoiceInfo.IssueDate),
+                SignatureValueFirmaFacturaAnterior = originalInvoiceInfo.Signature.Value
+            };
+        }
+
+        private static SoftwareFacturacionType1 Convert(Software software)
+        {
+            return new SoftwareFacturacionType1
+            {
+                LicenciaTBAI = software.License.Value,
+                Version = software.Version.Value,
+                Nombre = software.Name.Value,
+                EntidadDesarrolladora = Convert(software.Developer)
+            };
+        }
+
+        private static EntidadDesarrolladoraType1 Convert(Developer developer)
+        {
+            return developer.Match(
+                local => new EntidadDesarrolladoraType1
+                {
+                    Item = local.Nif.TaxpayerNumber
+                },
+                foreign => new EntidadDesarrolladoraType1
+                {
+                    Item = new IDOtro
+                    {
+                        CodigoPaisSpecified = foreign.Country.NonEmpty,
+                        CodigoPais = foreign.Country.Map(c => ConvertCountryType2(c)).ToNullable(),
+                        ID = foreign.Id.Value,
+                        IDType = Convert(foreign.IdType)
+                    }
+                }
+            );
+        }
+
+        private static CountryType2 ConvertCountryType2(Country country)
+        {
+            var result = country.Alpha2Code.ToEnum<CountryType2>();
+            return result.Get();
+        }
+
+        private static CountryType21 CovertCountryType21(Country country)
+        {
+            var result = country.Alpha2Code.ToEnum<CountryType21>();
+            return result.Get();
+        }
+
+        private static IDTypeType Convert(IdType type)
+        {
+            return type.Match(
+                IdType.NifVat, _ => IDTypeType.Item02,
+                IdType.Passport, _ => IDTypeType.Item03,
+                IdType.OfficialIdentificationDocumentIssuedByTheCountry, _ => IDTypeType.Item04,
+                IdType.ResidenceCertificate, _ => IDTypeType.Item05,
+                IdType.OtherSupportingDocument, _ => IDTypeType.Item06
+            );
+        }
+
+        private static IDTypeType1 Convert1(IdType type)
+        {
+            return type.Match(
+                IdType.NifVat, _ => IDTypeType1.Item02,
+                IdType.Passport, _ => IDTypeType1.Item03,
+                IdType.OfficialIdentificationDocumentIssuedByTheCountry, _ => IDTypeType1.Item04,
+                IdType.ResidenceCertificate, _ => IDTypeType1.Item05,
+                IdType.OtherSupportingDocument, _ => IDTypeType1.Item06
+            );
+        }
```

Comment:
I'm guessing there are some differences between these 2 enums, we're just not utilizing the differences, right?